File: Assets/Scripts/Spells/Spell.js
#pragma strict
class Spell {
	function Cast(playerTrans : Transform){
		Debug.Log('casting spell from: '+playerTrans.position.x+', '+playerTrans.position.y);
	}
	
	function CanLearn(player) {
		return true;
	}
	
	function CanCast(currentMP) {
		return true;
	}
}

class SpellInfo {
	var type : String = "Damage"; // Damage, Buff, Debuff
	var duration : float = 0; // 0 immediate/none, otherwise in seconds
	var damage : float = 0; // before defenses applied, after any plus mods
	var element : String; // fire, ice, water, etc for resistence checks
}

File: Assets/Scripts/Spells/Spell.cs
using UnityEngine;
using System.Collections;

public class Spell : Action
{

    override public bool OnFire(Transform playerTransform, Character character)
    {
        return true;
    }

    virtual public bool CanLearn(Character character)
    {
        return true;
    }

    virtual public bool CanCast(Character character)
    {
        return true;
    }

}
